src/modules/control_allocator/ControlAllocation/ControlAllocation.hpp

```diff
@@ -194,4 +205,7 @@ class ControlAllocation
 	matrix::Vector<float, NUM_ACTUATORS> _actuator_sp;  	//< Actuator setpoint
 	matrix::Vector<float, NUM_AXES> _control_sp;   		//< Control setpoint
 	matrix::Vector<float, NUM_AXES> _control_allocated;  	//< Allocated control
+
+	bool _A_update_needed{false};
```

Comment:
I am not sure about this. Control-allocation does not have a member _A because not all allocation methods need the pseudoinverse.
We can either
- copy _A and _A_update_needed and the associated logic in ControlAllocationSimple, ControlAllocationMultirotor and ControlAllocationSequentialDesaturation
- rename ControlAllocationSimple to ControlAllocationPseudoinverse and let ControlAllocationMultirotor and ControlAllocationSequentialDesaturation inherit from ControlAllocationPseudoinverse.- 
- do like you propose

I have a slight preference for option 2

Reply:
we will want to do some optimisation to avoid the computation of the pseudoinverse on the full 16x6 matrix, which will be easier with option 2